extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/AvroAttributeRowBuilder.java

```diff
@@ -0,0 +1,48 @@
+package mil.nga.giat.geowave.adapter.vector;
+
+import java.util.HashMap;
+
+import mil.nga.giat.geowave.core.index.ByteArrayId;
+import mil.nga.giat.geowave.core.index.StringUtils;
+import mil.nga.giat.geowave.core.store.adapter.NativeFieldHandler.RowBuilder;
+import mil.nga.giat.geowave.core.store.data.PersistentValue;
+
+import org.opengis.feature.simple.SimpleFeature;
+
+/**
+ * A GeoWave RowBuilder, used internally by AbstractDataAdapter to construct
+ * rows from a set field values (in this case SimpleFeatures from a set of
+ * attribute values). This implementation simply wraps a geotools
+ * SimpleFeatureBuilder.
+ * 
+ */
+public class AvroAttributeRowBuilder implements
+		RowBuilder<SimpleFeature, Object>
+{
+	private final HashMap<String, Object> idToValue = new HashMap<String, Object>();
+
+	private ByteArrayId id;
+
+	public AvroAttributeRowBuilder(
+			ByteArrayId byteArrayId ) {
+		this.id = byteArrayId;
+	}
+
+	@Override
+	public SimpleFeature buildRow(
+			final ByteArrayId dataId ) {
+		SimpleFeature sf = (SimpleFeature) idToValue.get(StringUtils.stringFromBinary(id.getBytes()));
+		return sf;
+	}
+
+	@Override
+	public void setField(
+			final PersistentValue<Object> fieldValue ) {
+		// Only interested in the relevant fields
+		if (id.equals(AvroFeatureAttributeHandler.FIELD_ID)) {
```

Comment:
Instead of using the class variable id for the comparison here, I think we should be comparing fieldValue.getId() to AvroFeatureAttributeHandler.FIELD_ID.  The class variable id is set when the AvroAttributeRowBuilder is constructed, and it looks like AvroFeatureAttributeHandler.FIELD_ID is passed, so this check will always evaluate to true.  I would just eliminate the id member variable as I don't think it's really needed.